lib/engines/hcktest/tests.rb

```diff
@@ -132,11 +132,37 @@ def wait_queued_test(id)
       end
     end
 
-    def test_parameters(test_name)
+    def read_test_config(test_name, config)
       @project.engine.config.tests_config
               .select { _1.tests.include?(test_name) }
-              .flat_map(&:parameters)
-              .to_h { |parameter| [parameter.name, parameter.value] }
+              .flat_map(&config)
+    end
+
+    def test_parameters(test_name)
+      read_test_config(test_name, :parameters)
+        .to_h { |parameter| [parameter.name, parameter.value] }
+    end
+
+    def run_guest_commands(test, commands)
+      return unless commands.guest_run
+
+      @logger.info("Running command (#{commands.desc}) on clients")
+      @tools.run_on_machine(@client.name, commands.desc, commands.guest_run)
+      @tools.run_on_machine(@support.name, commands.desc, commands.guest_run) if support_needed?(test)
+    end
+
+    def run_host_commands(commands)
+      return unless commands.host_run
+
+      @logger.info("Running command (#{commands.desc}) on host")
+      Process.wait spawn(*commands.host_run, out: :err)
```

Comment:
host commands can have replacements as in lib/setupmanagers/qemuhck/devices/tpm-tis.json for example

Reply:
Why don't use CmdRun class?

Reply:
CmdRun requires scope i just pick easiest way

Reply:
But who will control and terminate daemons for example in this case?